lib/languages/js/source-code/source-code.js

```diff
@@ -1164,14 +1187,19 @@ class SourceCode extends TokenStore {
      */
     finalize() {
 
-        // Step 1: ensure that all of the necessary variables are up to date
         const varsCache = this[caches].get("vars");
-        const globalScope = this.scopeManager.scopes[0];
         const configGlobals = varsCache.get("configGlobals");
         const inlineGlobals = varsCache.get("inlineGlobals");
         const exportedVariables = varsCache.get("exportedVariables");
+        const globalScope = this.scopeManager.scopes[0];
 
-        addDeclaredGlobals(globalScope, configGlobals, inlineGlobals);
+        /*
+         * For languages other than regular JavaScript, it's possible that there
+         * is no scope analysis done, and therefore, `globalScope` might be undefined.
+         */
+        if (globalScope) {
+            addDeclaredGlobals(globalScope, configGlobals, inlineGlobals);
+        }
```

Comment:
Also similar to https://github.com/eslint/eslint/pull/18448#discussion_r1609865923, and I also think we should rather update the tests to return a valid AST.